nakamoto-p2p/src/error.rs

```diff
@@ -27,4 +30,25 @@ pub enum Error {
 
     #[error("send event error: {0}")]
     SendEvent(String),
+
+    #[error("channel error: {0}")]
+    Channel(Box<dyn std::error::Error + Send + Sync>),
```

Comment:
This range of errors could be pushed down to the protocol module, so that the top-level error module doesn't need to know about crossbeam.